test/verifier.spec.js

```diff
@@ -1540,3 +1540,29 @@ test('invalid errorCacheTTL function should be handle ', async t => {
   clock.uninstall()
   t.end()
 })
+
+test('default errorCacheTTL should not cache errors when sub millisecond execution', async t => {
+  const clock = fakeTime({ now: 0 })
+
+  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjoxfQ.57TF7smP9XDhIexBqPC-F1toZReYZLWb_YRU5tv0sxM'
+  const verifier = createVerifier({
+    key: async () => {
+      throw new Error('invalid')
+    },
+    cache: true
+  })
+  const checkToken = 'check'
+  t.equal(verifier.cache.size, 0)
+  await t.rejects(async () => verifier(token))
+  t.equal(verifier.cache.size, 1)
```

Comment:
I'm not sure I understand this test. didn't we say that using the default value shouldn't cache errors? we do we expect to find one in the cache then?

Reply:
at the moment as long as the option `cache` is enabled everything goes into the cache regardless of the `TTL` even if it's 0
https://github.com/nearform/fast-jwt/blob/ba3640d10e4de293a3869f2739114c4b6faa73d5/src/verifier.js#L71-L90
I don't think this is the right behaviour but to change it requires to rework all tests because many are based on counting the elements in the cache which I think we can do it a future PR